src/service/core/workflow/tests/test_workflow_service_units.py

```
@@ -2017,6 +2019,93 @@ def test_backend_error_is_swallowed(self):
+
+    def __next__(self):
+        return next(self._lines)
+
+    def close(self):
+        self.closed = True
+
+
+class TestLogSourceRelease(unittest.IsolatedAsyncioTestCase):
+    """Covers release of the underlying log source when a client disconnects."""
+
+    _SCOPE = {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.4'}}
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)